persist/fs/mmap_linux.go

```diff
@@ -26,6 +26,18 @@ import (
 )
 
 func mmap(fd, offset, length int64, opts mmapOptions) ([]byte, error) {
+	// MAP_SHARED because we want writes to the byte slice to be reflected back in the underlying file
+	return mmapBase(fd, offset, length, syscall.MAP_SHARED, opts)
```

Comment:
Hm, so for file based mmap'ing we actually do not want to allow writes or to sync writes to the underlying file.

Maybe we need to rethink the abstractions to make that clear.

We should only ever mmap immutable data.

Reply:
Yeah I knew that, not sure why I left it the way it is. Writes will still fail because we're setting the read prop but not the write prop.

Anyways if we never want anything to propagate we should just use anonymous, not sure we need to rethink the abstractions here